panelapp/webservices/views.py

```diff
@@ -127,64 +127,68 @@ def get_panel(request, panel_name):
-        return Response(json)
-
+        queryset = GenePanelSnapshot.objects.get_active(all=True, internal=True, deleted=True)
     else:
         queryset = GenePanelSnapshot.objects.get_active()
 
-        queryset_name_exact = queryset.filter(panel__name=panel_name)
-        if not queryset_name_exact:
-            queryset_name = queryset.filter(panel__name__icontains=panel_name)
-            if not queryset_name:
-                queryset_old_names = queryset_name.filter(
-                    old_panels__icontains=panel_name
-                )
-                if not queryset_old_names:
+    queryset_name_exact = queryset.filter(panel__name=panel_name)
+    if not queryset_name_exact:
+        queryset_name = queryset.filter(panel__name__icontains=panel_name)
+        if not queryset_name:
+            queryset_old_names = queryset_name.filter(
+                old_panels__icontains=panel_name
+            )
+            if not queryset_old_names:
+                try:
                     try:
-                        try:
-                            int(panel_name)
-                            queryset_pk = queryset.filter(panel__pk=panel_name)
-                        except ValueError:
-                            queryset_pk = queryset.filter(panel__old_pk=panel_name)
-
-                        if not queryset_pk:
-                            return Response(
-                                {"Query Error: " + panel_name + " not found."}
-                            )
-                        else:
-                            queryset = queryset_pk
-                    except (DatabaseError, ValueError) as e:
-                        return Response({"Query Error: " + panel_name + " not found."})
-                else:
-                    queryset = queryset_old_names
+                        int(panel_name)
+                        queryset_pk = queryset.filter(panel__pk=panel_name)
+                    except ValueError:
+                        queryset_pk = queryset.filter(panel__old_pk=panel_name)
+
+                    if not queryset_pk:
+                        return Response(
+                            {"Query Error: " + panel_name + " not found."}
+                        )
+                    else:
+                        queryset = queryset_pk
+                except (DatabaseError, ValueError) as e:
+                    return Response({"Query Error: " + panel_name + " not found."})
             else:
-                queryset = queryset_name
+                queryset = queryset_old_names
         else:
-            queryset = queryset_name_exact
+            queryset = queryset_name
+    else:
+        queryset = queryset_name_exact
+
+    instance = queryset.first()
+
+    if version and instance:
+        if str(instance.major_version) != major_version and str(instance.minor_version) != minor_version:
+            snap = HistoricalSnapshot.objects.filter(
+                panel__pk=panel_name,
```

Comment:
`panel__pk` is an IntegerField, but people can use old mongodb 32hex to query the API - this will fail to find the snapshot. It's something that need to be fixed in the new API as well btw.

Reply:
yeah, I get 500 error when trying to access via old URL

https://bio-ote-panelapp.gel.zone/api/v1/panels/553f9696bb5a1616e5ed45d1/?version=1.2
https://bio-ote-panelapp.gel.zone/WebServices/get_panel/553f9696bb5a1616e5ed45d1/?version=1.2